### PR TITLE
feat(theme): manage security domain themes in the gateway

### PR DESCRIPTION
**Pull request by tcompiegne**
closes gravitee-io/issues#8125

**NOTE: Currently we can't change the color of the svg icons (eyes, arrows, ...)**

## :memo: Test scenarios 

1. Publish a theme with a custom color palette and a logo
2. Visit all the pages (login, reset password, registration, ...) and check if the theme has been applied

## :computer: Add screenshots for UI

theme-1
theme-2

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/manager/theme/ThemeManager.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class ThemeManager extends AbstractService implements InitializingBean, EventListener<ThemeEvent, Payload> {
```
Comment (farmborough) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/manager/theme/ThemeManager.java:41]: AbstractService missing its type parameter.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/vertx/view/thymeleaf/DomainBasedThemeResolver.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class DomainBasedThemeResolver {
+
+    private static String THEME_CONTEXT_KEY = "theme";
```
Comment (farmborough) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/vertx/view/thymeleaf/DomainBasedThemeResolver.java:33]: This constants should all be `static final`. Assuming they are constants, of course. If not, their names should be camelcase.

**Review by farmborough — CHANGES_REQUESTED** [review]
Just some trivial changes needed, otherwise LGTM.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/manager/theme/ThemeManager.java (hunk 1 above)
Comment (tcompiegne) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/manager/theme/ThemeManager.java]: FP

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/vertx/view/thymeleaf/DomainBasedThemeResolver.java (hunk 2 above)
Comment (tcompiegne) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/vertx/view/thymeleaf/DomainBasedThemeResolver.java]: FP

**Pull request merged**